string_grouper/string_grouper.py

```diff
@@ -651,6 +617,22 @@ def _validate_replace_na_and_drop(self):
                 "index if the number of index-levels does not equal the number of index-columns."
             )
 
+    @staticmethod
+    def _symmetrize_matrix(AA: csr_matrix) -> csr_matrix:
```

Comment:
This also adds a pep-8 warning because of the uppercase argument, but I get the naming convention of uppercase for a matrix. I think even scikit-learn does this. I was just wondering why the double AA for the input matrix? Is there a reason for that?

Another option could be X_non_symmetric for the input and X_symmetric for the output.

Reply:
I have no good reason for that.  I'll take your suggestion.